src/community/jdbcstore/src/main/resources/applicationContext.xml

```diff
@@ -25,7 +19,7 @@
   </bean>
   
   <!-- resource store -->
-  <bean id="jdbcResourceStore" class="org.geoserver.jdbcstore.JDBCResourceStoreFactoryBean">  
+  <bean id="resourceStoreImpl" class="org.geoserver.jdbcstore.JDBCResourceStoreFactoryBean">
```

Comment:
See what you mean about "resourceStoreImpl", can you call it resourceStoreFactory?

It feels like we are doing something wrong when we have to declare a factory, but yeah spring factory bean is apparently the way to go. I cannot see a consistent readable naming convention for factory beans.